File: ktor-server-netty/src/main/java/com/newrelic/labs/instrumentation/ktor/netty/KtorNettyHeaders.java
package com.newrelic.labs.instrumentation.ktor.netty;

import java.util.Collection;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.Headers;

public class KtorNettyHeaders implements Headers {
	
	io.ktor.http.Headers ktorHeaders = null;
	
	public KtorNettyHeaders(io.ktor.http.Headers kh) {
		ktorHeaders = kh;
	}

	@Override
	public HeaderType getHeaderType() {
		return HeaderType.HTTP;
	}

	@Override
	public String getHeader(String name) {
		return ktorHeaders.get(name);
	}

	@Override
	public Collection<String> getHeaders(String name) {
		return ktorHeaders.getAll(name);
	}

	@Override
	public void setHeader(String name, String value) {
		
	}

	@Override
	public void addHeader(String name, String value) {

	}

	@Override
	public Collection<String> getHeaderNames() {
		return ktorHeaders.names();
	}

	@Override
	public boolean containsHeader(String name) {
		return ktorHeaders.contains(name);
	}

}

File: ktor-server-netty-3.0/src/main/kotlin/com/newrelic/instrumentation/labs/ktor/netty/KtorNettyHeaders.kt
package com.newrelic.instrumentation.labs.ktor.netty

import com.newrelic.api.agent.HeaderType
import com.newrelic.api.agent.Headers
import java.util.Collections

class KtorNettyHeaders(private val ktorHeaders: io.ktor.http.Headers) : Headers {


    override fun getHeaderType(): HeaderType? {
        return HeaderType.HTTP
    }

    override fun getHeader(name: String): String? {
        val value = ktorHeaders.get(name)
        return value
    }

    override fun getHeaders(name: String): Collection<String?>? {
        val values = ktorHeaders.getAll(name)
        if (values != null) {
            return values
        } else {
            return Collections.emptyList()
        }
    }

    override fun setHeader(p0: String?, p1: String?) {
    }

    override fun addHeader(p0: String?, p1: String?) {
    }

    override fun getHeaderNames(): Collection<String?>? {
       return ktorHeaders.names()
    }

    override fun containsHeader(name: String): Boolean {
        return ktorHeaders.contains(name)
    }
}